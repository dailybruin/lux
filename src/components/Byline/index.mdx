---
name: Byline
route: /byline
---

import { Playground, PropsTable } from 'docz'
import Byline from '.'

# Byline

Every story needs an author and a byline.

<PropsTable of={Byline} />

<Playground>
  <Byline authors="Tanner Walters" />
  <Byline authors={['Tanner Walters', 'Mackenzie Possee', 'Jacob Preal']} />
</Playground>
