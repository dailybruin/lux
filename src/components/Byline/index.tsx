import * as React from 'react'
import { css } from 'react-emotion'
import toSentence from '../../utils/toSentence'

/**
 * Byline properties.
 */
interface BylineProps {
  /** The authors of the story. */
  authors: string | Array<string>
}

/** The byline of the story. */
export default function Byline(props: BylineProps) {
  return (
    <div className="byline">
      By{' '}
      {typeof props.authors === 'string'
        ? props.authors
        : toSentence(props.authors)}
    </div>
  )
}
