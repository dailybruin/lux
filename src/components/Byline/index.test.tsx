import * as React from 'react'
import { cleanup, render } from 'react-testing-library'
import Byline from '.'

describe('Byline', () => {
  afterEach(cleanup)

  it('Matches its snapshot', () => {
    const { container } = render(
      <Byline authors={['Nathan Smith', 'Michael Zhang']} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
