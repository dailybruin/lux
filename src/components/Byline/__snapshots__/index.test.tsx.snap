// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Byline Matches its snapshot 1`] = `
.emotion-0 {
  padding-top: 5px;
}

<div
  class="emotion-0"
>
  By 
  NATHAN SMITH and MICHAEL ZHANG
</div>
`;
